solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java

```diff
@@ -147,23 +150,30 @@ public DocTransformer create(String name, SolrParams localparams,
    *
    * @return a feature logger for the format specified.
    */
-  private FeatureLogger createFeatureLogger(String featureFormat) {
-    final FeatureLogger.FeatureFormat f;
-    if (featureFormat == null || featureFormat.isEmpty() ||
-        featureFormat.equals("sparse")) {
-      f = FeatureLogger.FeatureFormat.SPARSE;
-    }
-    else if (featureFormat.equals("dense")) {
-      f = FeatureLogger.FeatureFormat.DENSE;
-    }
-    else {
-      f = FeatureLogger.FeatureFormat.SPARSE;
-      log.warn("unknown feature logger feature format {}", featureFormat);
+  private FeatureLogger createFeatureLogger(String formatStr) {
+    final FeatureLogger.FeatureFormat format;
+    if (formatStr != null) {
+     format = getFeatureFormat(formatStr);
+     if (format == null) {
+       throw new IllegalArgumentException("unsupported feature format: "+format);
+     }
+    } else {
+      format = this.defaultFormat;
     }
     if (fvCacheName == null) {
       throw new IllegalArgumentException("a fvCacheName must be configured");
     }
-    return new CSVFeatureLogger(fvCacheName, f, csvKeyValueDelimiter, csvFeatureSeparator);
+    return new CSVFeatureLogger(fvCacheName, format, csvKeyValueDelimiter, csvFeatureSeparator);
+  }
+
+  private static FeatureLogger.FeatureFormat getFeatureFormat(String featureFormat) {
+    if (featureFormat.equals("sparse")) {
```

Comment:
minor: `FeatureLogger.FeatureFormat.valueOf(String) ? ` also, valueOf throws `IllegalArgumentException` if the string is not an enum value, maybe it is better than returning `null`?